### ISSUE TITLE
Load wrong version module with `require`

### ISSUE THREAD

**Issue by jhen0409** [rstacruz/pnpm#79]
I using `babel@6` and `fbjs-scripts` (it required babel 5), when I use pnpm install them, `babel-loader@6.x` will load `babel-core@5.x`, not `6.x`.

[This project](https://github.com/chentsulin/electron-react-boilerplate) can reproduce the problem. (try `npm run build`)

**Comment by rstacruz** [comment]
aw man, this is a side effect of the store layout, sadly. i guess it can be worked around but it won't be pretty

**Comment by rstacruz** [comment]
that project also fails to build electron-prebuilt under pnpm, still unsure why.

**Comment by jhen0409** [comment]
> that project also fails to build electron-prebuilt under pnpm, still unsure why.

I have also had electron-prebuilt problem, but I cannot be reproduce it. :\

**Comment by zkochan** [comment]
I think this should work from Node v6.3.0 if running node with the `--preserve-symlinks` flag

**Comment by zkochan** [comment]
I doesn't seem to be the initial problem, by I have this error in the specified project:

```
> electron-react-boilerplate@0.10.0 build /home/zkochan/src/test/electron-react-boilerplate
> npm run build-main && npm run build-renderer


> electron-react-boilerplate@0.10.0 build-main /home/zkochan/src/test/electron-react-boilerplate
> cross-env NODE_ENV=production node --preserve-symlinks -r babel-register ./node_modules/webpack/bin/webpack --config webpack.config.electron.js --progress --profile --colors

/home/zkochan/src/test/electron-react-boilerplate/node_modules/babel-register/node_modules/babel-core/lib/transformation/file/index.js:600
      throw err;
      ^

SyntaxError: /home/zkochan/.pnpm/.store/flat/webpack@1.13.3/_/bin/webpack.js: 'return' outside of function (12:2)
  10 | 	var localWebpack = require.resolve(path.join(process.cwd(), "node_modules", "webpack", "bin", "webpack.js"));
  11 | 	if(__filename !== localWebpack) {
> 12 | 		return require(localWebpack);
     | 		^
  13 | 	}
  14 | } catch(e) {}
  15 | var optimist = require("optimist")
    at Parser.pp$5.raise (/home/zkochan/src/test/electron-react-boilerplate/node_modules/babel-register/node_modules/babel-core/node_modules/babylon/lib/index.js:4215:13)
    at Parser.pp$1.parseReturnStatement (/home/zkochan/src/test/electron-react-boilerplate/node_modules/babel-register/node_modules/babel-core/node_modules/babylon/lib/index.js:1905:10)
    at Parser.pp$1.parseStatement (/home/zkochan/src/test/electron-react-boilerplate/node_modules/babel-register/node_modules/babel-core/node_modules/babylon/lib/index.js:1688:19)
    at Parser.pp$1.parseBlockBody (/home/zkochan/src/test/electron-react-boilerplate/node_modules/babel-register/node_modules/babel-core/node_modules/babylon/lib/index.js:2105:21)
    at Parser.pp$1.parseBlock (/home/zkochan/src/test/electron-react-boilerplate/node_modules/babel-register/node_modules/babel-core/node_modules/babylon/lib/index.js:2086:8)
    at Parser.pp$1.parseStatement (/home/zkochan/src/test/electron-react-boilerplate/node_modules/babel-register/node_modules/babel-core/node_modules/babylon/lib/index.js:1708:19)
    at Parser.pp$1.parseIfStatement (/home/zkochan/src/test/electron-react-boilerplate/node_modules/babel-register/node_modules/babel-core/node_modules/babylon/lib/index.js:1898:26)
    at Parser.pp$1.parseStatement (/home/zkochan/src/test/electron-react-boilerplate/node_modules/babel-register/node_modules/babel-core/node_modules/babylon/lib/index.js:1686:19)
    at Parser.pp$1.parseBlockBody (/home/zkochan/src/test/electron-react-boilerplate/node_modules/babel-register/node_modules/babel-core/node_modules/babylon/lib/index.js:2105:21)
    at Parser.pp$1.parseBlock (/home/zkochan/src/test/electron-react-boilerplate/node_modules/babel-register/node_modules/babel-core/node_modules/babylon/lib/index.js:2086:8)

```

It is probebly babel-register related

**Comment by zkochan** [comment]
In scripts, I added the `--preserve-symlinks` option to usages of node and I changed the address of webpack from `./node_modules/webpack/bin/webpack` to `./node_modules/.bin/webpack`. The resulting script section is this: 

```json
  "scripts": {
    "test": "cross-env NODE_ENV=test BABEL_DISABLE_CACHE=1 mocha --retries 2 --compilers js:babel-register --recursive --require ./test/setup.js test/**/*.spec.js",
    "test-watch": "npm test -- --watch",
    "test-e2e": "cross-env NODE_ENV=test BABEL_DISABLE_CACHE=1 mocha --retries 2 --compilers js:babel-register --require ./test/setup.js ./test/e2e.js",
    "lint": "eslint --ignore-path .gitignore --format=node_modules/eslint-formatter-pretty app test *.js",
    "hot-server": "cross-env NODE_ENV=development node --preserve-symlinks --max_old_space_size=2096 -r babel-register server.js",
    "build-main": "cross-env NODE_ENV=production node --preserve-symlinks -r babel-register ./node_modules/.bin/webpack --config webpack.config.electron.js --progress --profile --colors",
    "build-renderer": "cross-env NODE_ENV=production node --preserve-symlinks -r babel-register ./node_modules/.bin/webpack --config webpack.config.production.js --progress --profile --colors",
    "build": "npm run build-main && npm run build-renderer",
    "start": "cross-env NODE_ENV=production electron ./app/",
    "start-hot": "cross-env HOT=1 NODE_ENV=development electron -r babel-register -r babel-polyfill ./app/main.development",
    "postinstall": "concurrently \"install-app-deps\" \"node --preserve-symlinks node_modules/fbjs-scripts/node/check-dev-engines.js package.json\"",
    "dev": "npm run hot-server -- --start-hot",
    "package": "npm run build && build --publish never",
    "package-win": "npm run build && build --win --x64",
    "package-linux": "npm run build && build --linux",
    "package-all": "npm run build && build -mwl",
    "cleanup": "mop -v"
  }
```

Now the error is different: https://gist.github.com/zkochan/12e441a550e9cbeeff163e8846209a4c

The fix for this issue might be in #463

**Comment by zkochan** [comment]
With the latest pnpm@0.48 it works with no issues at all. There are two dev deps missing from electron-react-boilerplate, even thought they are used, so I created an issue there
